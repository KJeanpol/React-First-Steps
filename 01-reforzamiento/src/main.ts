import "./style.css";
import "./bases/11-async-await";


document.querySelector<HTMLDivElement>("#app")!.innerHTML = `
  <div>
    <H1>hOLA mUNDO</H1>
  </div>
`;
