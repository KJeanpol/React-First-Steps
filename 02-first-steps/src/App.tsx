import './App.css'

function App() {

  return (
    <>
      <h1>HOLA MUNDO</h1>
      <p>Hola, Pol</p>
    </>
  )
}

export default App
