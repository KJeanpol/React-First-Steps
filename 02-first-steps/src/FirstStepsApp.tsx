// App punto de entrada de la aplicacion

import { ItemCounter } from "./shopping-cart/ItemCounter";

interface ItemInCart {
  productName: string;
  quantity: number;
}

const itemsIncart: ItemInCart[] = [
  { productName: "Nintendo Switch 2", quantity: 3 },
  { productName: "PS5", quantity: 5 },
  { productName: "Control Xbox", quantity: 9 },
];

export function FirstStepsApp() {
  return (
    <>
      <h1>Carrito de Compras</h1>

      {/* con desestructuracion se veria itemsIncart.map(({productname, quantity})*/}
      {itemsIncart.map((item) => (
        <ItemCounter
          key={item.productName} //la key siempre debe ser unica
          name={item.productName}
          quantity={item.quantity}
        />
      ))}
      {/* <ItemCounter name="Nintendo Switch 2" quantity={3} />
      <ItemCounter name="PS5" quantity={3} />
      <ItemCounter name="PC Gamer" quantity={3} /> */}
    </>
  );
}
