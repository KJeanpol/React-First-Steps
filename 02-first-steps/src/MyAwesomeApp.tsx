// Si son constantes es mejor definirlas fuera del componente
// para evitar que se redefinan en cada render (revisar si hay cambios)
// en el ciclo de revision y ejecucion de React

import type { CSSProperties } from "react";

const name = "Jeanpol";
const lastName = "Alvarado";
const favoriteGames = ["Zelda", "Mario", "Metroid"];
//valores booleanos no generan ningun cambio en el DOM
const isActive = false;
const address = {
  street: "123 Main St",
  city: "Anytown",
  country: "USA",
};

const myStyles: CSSProperties = {
  color: "red",
  backgroundColor: "black",
  fontWeight: "bold",
};

// style = {{}}  doble llave porque es un objeto de JSX
// la primera llave es para indicar que es JS y la segunda es el objeto
export function MyAwesomeApp() {
  return (
    <div>
      <h1>{name}</h1>
      <h3>{lastName}</h3>
      <p className="mi-clase-favorita">{favoriteGames.join(", ")}</p>
      <p style={myStyles}>{isActive ? "Activo" : "Inactivo"}</p>
      <p>{JSON.stringify(address)}</p>
    </div>
  );
}
