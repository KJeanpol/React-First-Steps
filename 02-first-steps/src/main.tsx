import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import "./index.css";
import { FirstStepsApp } from "./FirstStepsApp";
// import { MyAwesomeApp } from "./MyAwesomeApp";
// import App from './App.tsx'

createRoot(document.getElementById("root")!).render(
  // <StrictMode>
  //   <App />
  // </StrictMode>,
  // asegurarse que la aplicacion funcione como esperamos

  <StrictMode>
    <FirstStepsApp />
    {/* <MyAwesomeApp /> */}
  </StrictMode>
);
