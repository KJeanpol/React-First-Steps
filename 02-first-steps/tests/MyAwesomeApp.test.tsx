import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "../src/MyAwesomeApp";
import { render, screen } from "@testing-library/react";
describe("My Awesome App Test Suit", () => {
  test("Should render firstName and lastName with RENDER", () => {
    //vitest crea un DOM virtual con jsdom
    // render es para pruebas de snapshot o pruebas donde se requiera inspeccionar el DOM
    // 1. arrange

    const { container } = render(<MyAwesomeApp />);

    // 2. act
    const h1 = container.querySelector("h1");
    console.log(h1?.innerHTML);

    // 3. assert
    expect(h1?.innerHTML).toContain("Jeanpol");
  });

  test("Should render firstName and lastName with SCREEN", () => {
    //vitest crea un DOM virtual con jsdom
    // screen es conveniente para pruebas donde se requiera interaccion

    // 1. arrange
    render(<MyAwesomeApp />);
    screen.debug();
    // 2. act
    const h1 = screen.getByRole("heading", { level: 1 });
    const h3 = screen.getByRole("heading", { level: 3 });

    // 3. assert
    expect(h1.innerHTML).toContain("Jeanpol");
  });

  test("Should match Snapshot with RENDER", () => {
    // es como poner un candado a la estructura HTML
    // 1. arrange
    const { container } = render(<MyAwesomeApp />);

    screen.debug();
    // 2. act

    // 3. assert
    expect(container).toMatchSnapshot();
  });
});
