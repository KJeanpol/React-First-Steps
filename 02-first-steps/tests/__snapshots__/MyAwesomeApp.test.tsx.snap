// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Awesome App Test Suit > Should match Snapshot with RENDER 1`] = `
<div>
  <div>
    <h1>
      Jeanpol
    </h1>
    <h3>
      Alvarado
    </h3>
    <p
      class="mi-clase-favorita"
    >
      Zelda, Mario, Metroid
    </p>
    <p
      style="color: red; background-color: black; font-weight: bold;"
    >
      Inactivo
    </p>
    <p>
      {"street":"123 Main St","city":"Anytown","country":"USA"}
    </p>
  </div>
</div>
`;
